addons/website_event_sale/models/sale_order.py

```diff
@@ -30,7 +30,9 @@ def _website_product_id_change(self, order_id, product_id, qty=0, **kwargs):
         else:
             product = self.env['product.product'].browse(product_id)
             if product.event_ticket_ids:
-                event_ticket_id = product.event_ticket_ids[0].id
+                # When a customer adds an event product directly to the cart we don't want to assign tickets from
+                # ended events as it's confusing and leads to errors.
+                event_ticket_id = product.event_ticket_ids.filtered(lambda t: not t.event_id.stage_id.pipe_end)[0:].id
```

Comment:
Not sure if just a typo but if the column was intended it should be `[:1]` not `[0:]`
Also could remove the 'if' in that case as it's redundant with the filter and the `[:1]`

Also wouldn't it make sense to just use `even_id.is_finished` instead? Seems more idiomatic, pipe_end doesn't necessarily imply buying a ticket doesn't make sense anymore I think

Reply:
Sure, I wanted to avoid failing on an empty recordset but that's not the way to do it for sure :smile:  

I took your idea for an event more semantic way using `is_ongoing` and `fields.first`